src/Bootstrappers/FilesystemTenancyBootstrapper.php

```diff
@@ -54,20 +54,20 @@ public function bootstrap(Tenant $tenant)
         }
 
         // Storage facade
+        Storage::forgetDisk($this->app['config']['tenancy.filesystem.disks']);
+
         foreach ($this->app['config']['tenancy.filesystem.disks'] as $disk) {
-            /** @var FilesystemAdapter $filesystemDisk */
-            $filesystemDisk = Storage::disk($disk);
-            $this->originalPaths['disks'][$disk] = $filesystemDisk->getAdapter()->getPathPrefix();
+            $originalRoot = $this->app['config']["filesystems.disks.{$disk}.root"];
+            $this->originalPaths['disks'][$disk] = $originalRoot;
 
-            if ($root = str_replace(
+            $finalPrefix = str_replace(
                 '%storage_path%',
                 storage_path(),
-                $this->app['config']["tenancy.filesystem.root_override.{$disk}"] ?? ''
-            )) {
-                $filesystemDisk->getAdapter()->setPathPrefix($finalPrefix = $root);
-            } else {
-                $root = $this->app['config']["filesystems.disks.{$disk}.root"];
-                $filesystemDisk->getAdapter()->setPathPrefix($finalPrefix = $root . "/{$suffix}");
+                $this->app['config']["tenancy.filesystem.root_override.{$disk}"] ?? '',
+            );
+
```

Comment:
```suggestion

$finalPrefix = str_replace('%tenant%', $suffix, $finalPrefix);

```
I would like to suggest the following suggestion if this is accepted.

Reply:
If this is a new feature, like in the other PR, it's best to add it after merging this.

This PR should be just for the Laravel upgrade